src/Printer/Console.php

```diff
@@ -4,52 +4,83 @@
 {
     private const DEFAULT_LINE_LENGTH = 80;
 
-    public function printData(OutputInterface $output, FileReportList $fileReportList, HintList $hintList): void
+    /**
+     * @var Highlighter
+     */
+    private $highlighter;
+
+    public function __construct(Highlighter $highlighter)
+    {
+        $this->highlighter = $highlighter;
+    }
+
+    public function printData(OutputInterface $output, HintList $hintList, array $list): void
     {
         $length = (int) (`tput cols` ?: self::DEFAULT_LINE_LENGTH);
         $separator = str_repeat('-', $length);
         $output->writeln(PHP_EOL . $separator . PHP_EOL);
 
-        $total = 0;
-        foreach ($fileReportList->getFileReports() as $fileReport) {
-            $entries = $fileReport->getEntries();
-            $total += count($entries);
-            foreach ($entries as $entry) {
+        foreach ($this->groupDetectionResultPerFile($list) as $detectionResults) {
+            foreach ($detectionResults as $detection) {
                 $output->writeln(sprintf(
-                    '%s:%d  Magic number: %s',
-                    $fileReport->getFile()->getRelativePathname(),
-                    $entry['line'],
-                    $entry['value']
+                    '%s:%d. Magic number: %s',
+                    $detection->getFile()->getRelativePathname(),
+                    $detection->getLine(),
+                    $detection->getValue()
                 ));
 
-                $highlighter = new Highlighter(new ConsoleColor());
                 $output->writeln(
-                    $highlighter->getCodeSnippet($fileReport->getFile()->getContents(), $entry['line'], 0, 0)
+                    $this->highlighter->getCodeSnippet(
+                        $detection->getFile()->getContents(),
+                        $detection->getLine(),
+                        0,
+                        0
+                    )
                 );
 
                 if ($hintList->hasHints()) {
-                    $hints = $hintList->getHintsByValue($entry['value']);
-                    if (false === empty($hints)) {
+                    $hints = $hintList->getHintsByValue($detection->getValue());
+
+                    if ($hints !== []) {
                         $output->writeln('Suggestions:');
+
                         foreach ($hints as $hint) {
                             $output->writeln("\t\t" . $hint);
                         }
+
                         $output->write(PHP_EOL);
                     }
                 }
             }
+
             $output->writeln($separator . PHP_EOL);
         }
-        $output->writeln('<info>Total of Magic Numbers: ' . $total . '</info>');
+
+        $output->writeln('<info>Total of Magic Numbers: ' . count($list) . '</info>');
+    }
+
+    /**
+     * @param array<int, DetectionResult> $list
+     *
+     * @return array<int, DetectionResult[]>
+     */
+    private function groupDetectionResultPerFile(array $list): array
```

Comment:
same here, $list could be $foundMagicNumbers or similar

Reply:
done